src/request/creditcard.spec.ts

```diff
@@ -0,0 +1,54 @@
+import { ArgumentError } from '../errors';
+import CreditCard from './creditcard';
+
+describe('CreditCard()', () => {
+  it('throws an error if avsResult is not valid', () => {
+    expect(() => {
+      const creditcard = new CreditCard({
+        avsResult: 'foo',
+      });
+    }).toThrowError(ArgumentError);
```

Comment:
It might be good to test that the `ArguementError` has a specific value. Currently, `avsResult` does throw the `ArgumentError`, but you aren't for certain where in your code you're testing against. In theory, the `singleChar` regex test could have validated unexpectedly, the script continues to the next condition statement, and then `ArguementError` is thrown for other reasons.

I think these tests can be rewritten something like this:
```
    const avsResult = 'foo';
    const creditcard = () => new CreditCard({
      avsResult,
    });

    expect(creditcard).toThrowError(ArgumentError);
    expect(creditcard).toThrowError(avsResult);
```